internal/resource/configvalidators/char_validator.go

```diff
@@ -0,0 +1,43 @@
+package configvalidators
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+var _ validator.String = &charValidator{}
+
+type charValidator struct{}
+
+func (v charValidator) Description(ctx context.Context) string {
+	return "The ID must be less than 33 characters, contain no spaces, and be alphanumeric"
```

Comment:
The name of this validator doesn't really align with this description. We should either make the name more specific or add some parameters to the validator to choose how many characters for example

Reply:
edited this file completely with more info